lightningd/peer_control.c

```diff
@@ -894,7 +895,10 @@ peer_connected_hook_cb(struct peer_connected_hook_payload *payload,
 		abort();
 	}
 
-	notify_connect(ld, &peer->id, &addr);
+	noti_payload = tal(tmpctx, struct connect_notification_payload);
```

Comment:
Not sure this needs to be allocated on the heap at all, if we immediately call the notification-serialization and pass that to the subscribers (if there are any), we retain control of the struct and can leave it on the stack.

Reply:
Would save us one allocation :-)

Reply:
Thank you! I'll try to take all struct fields as arguments for function calls. :-)